### DISCUSSION TITLE
Reason for Older Deno Version on Homebrew

### DISCUSSION THREAD

**Discussion by ben-laird** [denoland/deno#25540] [Q&A] [answered]
Hello,

I use Deno through Homebrew, and because of that, Deno does not allow me to upgrade when a new version is available. This is perfectly ok, I can understand why this is the case. Besides, it just forces me to update all my Homebrew taps, which I should be doing anyway. My only (somewhat small) gripe is the Homebrew version tends to lag behind the latest version of Deno by a couple patch versions to an entire minor version. As of writing, `1.45.5` is the latest Homebrew will allow, while the current version is `1.46.3`.

I'm assuming there's a reason for this that I just don't know and that this is not a bug in some deployment pipeline, so I'd like to just ask the question here for my own information and for future reference if someone else has this question too. Thanks for taking the time to answer.

**Comment by bartlomieju** [comment, marked as answer]
It's because Homebrew was not able to build Deno properly. We provided a solution here: https://github.com/denoland/deno/issues/25298 but the PR was not merged. We'll look into opening another PR.

**Reply by ben-laird** [reply]
Great to hear, thanks so much for working on it. How often does Homebrew fail to release Deno? I read in the issue you linked that Homebrew actually cuts the release, and you have no control over when they do so. I've also seen the version lag a couple times before, likely for the same reason, so I'd just like to set my expectations and not rush you guys. Thanks again for your reply!

**Reply by bartlomieju** [reply]
> How often does Homebrew fail to release Deno?

In the past 3 months this has been a recurring theme on every minor release.

> I've also seen the version lag a couple times before, likely for the same reason, so I'd just like to set my expectations and not rush you guys.

I strongly advise to use the installation script available on https://deno.com and using `deno upgrade` to update to the latest version. It's just a lot more convenient and faster.

**Comment by bartlomieju** [comment]
FYI Deno v1.46.3 is now available on Homebrew: https://formulae.brew.sh/formula/deno#default
